File: codigos/LeitorCSV.java
package codigos;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class LeitorCsv {

    public static List<Partida> caminhoCsv;

    public static List<Partida> carregarPartidas(String caminho, Map<String, Time> times) {
        List<Partida> partidas = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new FileReader(caminho, StandardCharsets.UTF_8))) {
            String linha = br.readLine();

            while ((linha = br.readLine()) != null) {
                if (linha.isBlank()) continue;

                String[] dados = linha.split(",", -1); 
                if (dados.length < 16) continue;

                for (int i = 0; i < dados.length; i++) {
                    dados[i] = dados[i].replace("\"", "").trim();
                }

                int id = Integer.parseInt(dados[0]); 
                String mandante = dados[4];
                String visitante = dados[5];
                String estadio = dados[11];
                String data = dados[2];
                String hora = dados[3];
                String vencedor = "-".equals(dados[10]) ? "Empate" : dados[10];
                int placarMandante = Integer.parseInt(dados[12]);
                int placarVisitante = Integer.parseInt(dados[13]);
                String estadoMandante = dados[14];
                String estadoVisitante = dados[15];

                Partida p = new Partida(
                        id,
                        mandante,
                        visitante,
                        estadio,
                        data,
                        hora,
                        vencedor,
                        placarMandante,
                        placarVisitante
                );
                partidas.add(p);

                times.putIfAbsent(mandante, new Time(mandante, "", estadoMandante));
                times.putIfAbsent(visitante, new Time(visitante, "", estadoVisitante));

            }

        } catch (IOException e) {
            e.printStackTrace();
        }

        return partidas;
    }
}

File: Python/codigosPython/LeitorCSV.py
import csv
from typing import List, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from .Time import Time
    from .Partida import Partida


class LeitorCSV:
    @staticmethod
    def carregarPartidas(caminho: str, times: Dict[str, "Time"]) -> List["Partida"]:
        from .Partida import Partida
        from .Time import Time

        partidas: List[Partida] = []

        with open(caminho, encoding="utf-8") as f:
            leitor = csv.reader(f)
            next(leitor, None)  

            for linha in leitor:
                if not linha or len(linha) < 16:
                    continue

                dados = [col.replace('"', '').strip() for col in linha]

                mandante = dados[4]
                visitante = dados[5]
                estadio = dados[11]
                data = dados[2]
                hora = dados[3]
                vencedor = dados[10]

                try:
                    placarMandante = int(dados[12])
                    placarVisitante = int(dados[13])
                except ValueError:
                    continue

                estadoMandante = dados[14]
                estadoVisitante = dados[15]

                p = Partida(
                    mandante=mandante,
                    visitante=visitante,
                    estadio=estadio,
                    data=data,
                    hora=hora,
                    vencedor=vencedor,
                    placarMandante=placarMandante,
                    placarVisitante=placarVisitante,
                )
                partidas.append(p)

                if mandante not in times:
                    times[mandante] = Time(nome=mandante, tecnico="", estado=estadoMandante)
                if visitante not in times:
                    times[visitante] = Time(nome=visitante, tecnico="", estado=estadoVisitante)

                times[mandante].atualizarEstatisticas(p)
                times[visitante].atualizarEstatisticas(p)

        return partidas
